src/main/java/org/usfirst/frc4079/RobotBuilderProject1/commands/GamepadRunIntake.java

```diff
@@ -0,0 +1,56 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2018 FIRST. All Rights Reserved.                             */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+package org.usfirst.frc4079.RobotBuilderProject1.commands;
+
+import org.usfirst.frc4079.RobotBuilderProject1.Robot;
+
+import edu.wpi.first.wpilibj.command.Command;
+
+public class GamepadRunIntake extends Command {
+  public GamepadRunIntake() {
+    // Use requires() here to declare subsystem dependencies
+    // eg. requires(chassis);
+    requires(Robot.intake);
+  }
+
+  // Called just before this Command runs the first time
+  @Override
+  protected void initialize() {
+  }
+
+  // Called repeatedly when this Command is scheduled to run
+  @Override
+  protected void execute() {
+    if(Robot.oi.gamePad.getRightBumper()) {
+        Robot.intake.run(0.4);
+    } else if(Robot.oi.gamePad.getLeftBumper()) {
+        Robot.intake.run(-0.3);
+    } else {
+      Robot.intake.run(0);
+    }  
+  }
+
+  // Make this return true when this Command no longer needs to run execute()
+  @Override
+  protected boolean isFinished() {
+    return false;
+  }
+
+  // Called once after isFinished returns true
+  @Override
+  protected void end() {
+    Robot.intake.stop();
```

Comment:
Hmmm, the comment above this method says that this will be called only when `isFinished` returns true, which it never does (line 41). This line is never reached. Is setting the intake power to 0 sufficient for stopping the intake?

Reply:
I think that isFinished() only returns false because it's suppose to start in teleopinit() and runs for the entirety of the teleop period (basically the rest of the match). In this case, end() may be irrevelant for its usual purpose unless end() is called when the match ends.